launcher/src/main/java/org/apache/spark/launcher/LauncherServer.java

```diff
@@ -337,6 +337,10 @@ public void close() throws IOException {
       }
       super.close();
       if (handle != null) {
+    	if (!handle.getState().isFinal()) {
+    	  LOG.log(Level.WARNING, "Lost connection to launch server.");
```

Comment:
minor: this is running in the server, so you lost the connection to the Spark application.